core/services/synchronization/stats_pusher.go

```diff
@@ -60,54 +68,51 @@ const (
 	updateCallbackName = "sync:run_after_update"
 )
 
-// NewStatsPusher returns a new event queuer
-func NewStatsPusher(orm *orm.ORM, url *url.URL, accessKey, secret string, afters ...utils.Afterer) StatsPusher {
+// NewStatsPusher returns a new StatsPusher service
+func NewStatsPusher(orm *orm.ORM, explorerClient ExplorerClient, afters ...utils.Afterer) StatsPusher {
 	var clock utils.Afterer
 	if len(afters) == 0 {
 		clock = utils.Clock{}
 	} else {
 		clock = afters[0]
 	}
 
-	sp := &statsPusher{
-		ORM:      orm,
-		WSClient: noopWebSocketClient{},
-		Period:   30 * time.Minute,
-		clock:    clock,
+	return &statsPusher{
+		ORM:            orm,
+		ExplorerClient: explorerClient,
+		Period:         30 * time.Minute,
+		clock:          clock,
 		backoffSleeper: backoff.Backoff{
 			Min: 1 * time.Second,
 			Max: 5 * time.Minute,
 		},
 		waker: make(chan struct{}, 1),
 	}
-
-	if url != nil {
-		sp.WSClient = NewWebSocketClient(url, accessKey, secret)
-		gormCallbacksMutex.Lock()
-		_ = orm.RawDB(func(db *gorm.DB) error {
-			db.Callback().Create().Register(createCallbackName, createSyncEventWithStatsPusher(sp, orm))
-			db.Callback().Update().Register(updateCallbackName, createSyncEventWithStatsPusher(sp, orm))
-			return nil
-		})
-		gormCallbacksMutex.Unlock()
-	}
-	return sp
 }
 
+// GetURL returns the URL where stats are being pushed
 func (sp *statsPusher) GetURL() url.URL {
-	return sp.WSClient.Url()
+	return sp.ExplorerClient.Url()
 }
 
+// GetStatus returns the ExplorerClient connection status
 func (sp *statsPusher) GetStatus() ConnectionStatus {
-	return sp.WSClient.Status()
+	return sp.ExplorerClient.Status()
 }
 
 // Start starts the stats pusher
 func (sp *statsPusher) Start() error {
-	err := sp.WSClient.Start()
+	gormCallbacksMutex.Lock()
```

Comment:
Mutex never gets unlocked if we return error

Reply:
See https://github.com/smartcontractkit/chainlink/pull/3400#discussion_r509466874

Reply:
Fixed.